calculate_trigger.py

```diff
@@ -52,9 +74,39 @@ def search_for_fault_location(
         goldenrun_tb_info.at[idtbinfo, "assembler"],
         goldenrun_tb_info.at[idtbinfo, "id"],
     )
+    return [idx, ins]
+
+
+def search_for_fault_location(
+    filter_lists,
+    trigger_position,
+    fault_address,
+    trigger_occurrences,
+    fault_lifespan,
+    goldenrun_tb_exec,
+    goldenrun_tb_info,
+):
+    logger.info(f"Search trigger to fault INSN at 0x{fault_address:08x}")
+    [idx, ins] = find_fault(
+        fault_address, goldenrun_tb_exec, goldenrun_tb_info, trigger_occurrences
+    )
+    if idx < 0:
+        return [-1, trigger_occurrences, fault_lifespan]
     trigger_not_in_same_tb = 0
+    lifespan_diff = trigger_position + fault_lifespan
```

Comment:
Please handle the case when `lifespan_diff` is negative. The fault is invalid in this case and we should print a warning.

Reply:
If you just return with a negative trigger address, an error should be printed when the fault is removed. So you don't have to print a warning here.

Reply:
It would be good to give a warning and set negative trigger address here i think, just to make it transparent to the user why the fault was not executed.